lib/worker.js

```diff
@@ -15,7 +15,17 @@ const state = new Int32Array(stateBuf)
 const data = Buffer.from(dataBuf)
 
 async function start () {
-  const fn = (await import(workerData.filename)).default
+  let fn
+  try {
+    fn = (await import(workerData.filename)).default
+  } catch (error) {
+    if ((error.code === 'ENOTDIR' || error.code === 'ERR_MODULE_NOT_FOUND') &&
+     workerData.filename.startsWith('file://')) {
+      fn = require(workerData.filename.replace('file://', ''))
```

Comment:
Can you please add a few comments explaining why all of this is needed? Add links to issues as well. I suspect if somebody looked at this would have questions.